src/main/java/spoon/support/reflect/declaration/CtTypeImpl.java

```diff
@@ -765,18 +765,18 @@ private boolean isSameParameter(CtMethod<?> method, CtTypeReference<?> ctParamet
 				return false;
 			}
 		} else if (expectedType instanceof CtTypeParameterReference) {
-			if (!ctParameterType.isSubtypeOf(factory.Type().createReference(expectedType.getActualClass()))) {
+			if (!ctParameterType.isSubtypeOf(factory.Type().createReference(expectedType.getQualifiedName()))) {
```

Comment:
A more stable approach could be the check if `getDeclaration` returns null and only if it is not null using `getQualifiedName()` to create the reference. Otherwise, skip this check.

Reply:
getActualClass should be avoided as much as possible to have portable code between full-classpath and noclasspath mode.
So this change seems good to me.